userguide/vmware.rst

```diff
@@ -0,0 +1,133 @@
+.. index:: VMware Recommendations
+.. _VMware Recommendations:
+
+VMware Recommendations
+======================
+
+This section offers %brand% configuration recommendations and
+troubleshooting tips when using %brand% with a
+`VMware <https://www.vmware.com/>`__ hypervisor.
+
+
+.. _VMware guest:
+
+%brand% as a VMware Guest
+-----------------------------------
+
+This section has recommendations for configuring %brand% when it is
+installed as a Virtual Machine (VM) in VMware.
+
+#ifdef freenas
+To create a new %brand% Virtual Machine (VM) in VMware, see the
+:ref:`VMware ESXi` section of this guide.
+#endif freenas
+
+When upgrading %brand%, be aware that the VMware Tools VMXNET3 drivers
+are not supported. Configure and use the
+`vmx(4) <https://www.freebsd.org/cgi/man.cgi?query=vmx>`__ driver
+instead.
+
+Network connection errors for plugins or jails inside the %brand% VM can
+be caused by a misconfigured
+`virtual switch <https://pubs.vmware.com/vsphere-51/index.jsp?topic=%2Fcom.vmware.wssdk.pg.doc%2FPG_Networking.11.4.html>`__
+or
+`VMware port group <https://pubs.vmware.com/vsphere-4-esx-vcenter/index.jsp?topic=/com.vmware.vsphere.server_configclassic.doc_40/esx_server_config/networking/c_port_groups.html>`__.
+Make sure MAC spoofing and promiscuous mode are enabled on the switch
+first, and then the port group the VM is using.
+
+
+.. _Hosting Storage:
+
+Using %brand% to Host Storage for VMware
+--------------------------------------------------
+
+This section has recommendations for configuring %brand% when the system
+is being used as a VMware datastore.
+
+#ifdef truenas
+
+ALUA **must** be used when configuring
+:ref:`iSCSI Sharing <Block (iSCSI)>` for a %brand% with
+High Availability (HA). This allows %brand% to
+:ref:`failover <Failover>` when there is an active iSCSI share.
+
+#endif truenas
+
+Change the VM disk timeouts to prevent corruption of VM guest file
+systems.
+#ifdef truenas
+This also helps VMs deal with %brand% High Availability (HA)
+:ref:`failovers <Failover>`.
+#endif truenas
+Set the timeout to a minimum of *180 seconds*. See the guest operating
+system documentation for setting disk timeouts. VMware provides
+instructions for setting disk timeouts on some specific guest operating
+systems:
+
+* Windows guest operating system:
+  `<https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.storage.doc/GUID-EA1E1AAD-7130-457F-8894-70A63BD0623A.html>`__
+
+* Linux guests running kernel version *2.6*:
+  `<https://kb.vmware.com/s/article/1009465>`__
+
+When creating a new Virtual Machine (VM), be sure to install the VMware
```

Comment:
I'm not clear on the point of this.  Maybe originally it applied to running FreeNAS as a VMware guest.